internal/jobobject/queue.go

```diff
@@ -0,0 +1,96 @@
+package jobobject
```

Comment:
This seems pretty standalone and not job object specific. I think this should be moved to it's own package.

Reply:
The only use case for it currently is for handling the iocp messages. If we can think of another case I'd be fine moving it out but for now I'd just prefer to keep it

Reply:
I see what you're saying, but given how generic it is, it doesn't really fit going in a package that is specific to job object needs. I'd rather it be moved out and the highlevel API be exported. If not for code dup then at least for making the repo's packages more logically organized.

Reply:
Moved to new package